File: Java/Easy/1886 Determine Whether Matrix Can Be Obtained By Rotation.java
class Solution {
    public boolean findRotation(int[][] mat, int[][] target) {
        int n = mat.length;
        
        // Assume all 4 possible rotations could be valid initially
        boolean deg0 = true;
        boolean deg90 = true;
        boolean deg180 = true;
        boolean deg270 = true;
        
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < n; j++) {
                // Check 0 degrees (no rotation)
                if (mat[i][j] != target[i][j]) {
                    deg0 = false;
                }
                // Check 90 degrees clockwise
                if (mat[i][j] != target[j][n - 1 - i]) {
                    deg90 = false;
                }
                // Check 180 degrees
                if (mat[i][j] != target[n - 1 - i][n - 1 - j]) {
                    deg180 = false;
                }
                // Check 270 degrees clockwise
                if (mat[i][j] != target[n - 1 - j][i]) {
                    deg270 = false;
                }
            }
        }
        
        // If any of the rotations remain true, then it's a valid match
        return deg0 || deg90 || deg180 || deg270;
    }
}

File: Python/Easy/1886 Determine Whether Matrix Can Be Obtained By Rotation.py
class Solution:
    def findRotation(self, mat: list[list[int]], target: list[list[int]]) -> bool:
        n = len(mat)
        
        # Assume all 4 possible rotations could be valid initially
        deg0 = deg90 = deg180 = deg270 = True
        
        for i in range(n):
            for j in range(n):
                # Check 0 degrees (no rotation)
                if mat[i][j] != target[i][j]:
                    deg0 = False
                
                # Check 90 degrees clockwise
                if mat[i][j] != target[j][n - 1 - i]:
                    deg90 = False
                
                # Check 180 degrees
                if mat[i][j] != target[n - 1 - i][n - 1 - j]:
                    deg180 = False
                
                # Check 270 degrees clockwise
                if mat[i][j] != target[n - 1 - j][i]:
                    deg270 = False
                    
        # If any rotation remains True, it's a valid match
        return deg0 or deg90 or deg180 or deg270
